src/attribute.c

```diff
@@ -23,3 +23,100 @@ void SetPlayerName(u16 *nameBuffer)
     }
 }
 
+void sub_08017B7C(u16 *buffer)
+{
+    struct SaveBlock1 *sb1 = gSaveBlock1Ptr;
+    u16 i = 0;
+
+    sb1->unk16[0] = buffer[0];
+    while (buffer[i])
+    {
+        ++i;
+        sb1->unk16[i] = buffer[i];
+    }
+}
+/* The structure of this func is the same to the upper one. But after checking the memory, unk16 is not where the name array lays. */
```

Comment:
Same issue. Please move it above the function. 
And the comment itself is very verbose, you can just say something like `The same as SetPlayerName but uses unk16.` .